### PR TITLE
delete/reset/stop projections

### PR DESCRIPTION
**Pull request by prolic**
resolves https://github.com/prooph/pdo-event-store/issues/50

### CODE REVIEW

**Comment by prolic** [comment]
this one is really nice for event-store-http-api !!!

File: src/MySqlEventStore.php [hunk 1]
```diff
@@ -490,13 +499,59 @@ public function getDefaultReadModelProjectionFactory(): ReadModelProjectionFacto
...
         }
 
         return $this->defaultReadModelProjectionFactory;
     }
 
+    public function deleteProjection(string $name, bool $deleteEmittedEvents): void
+    {
+        $sql = <<<EOT
+UPDATE $this->projectionsTable SET status = ? WHERE name = ? LIMIT 1;
+EOT;
```
Comment (sandrokeil) [inline src/MySqlEventStore.php]: Why is `LIMIT 1` here used and not in the `PostgresEventStore` class?

File: src/Projection/PdoEventStoreReadModelProjection.php [hunk 2]
```diff
@@ -368,19 +407,37 @@ public function run(bool $keepRunning = true): void
...
+
+                return;
+            case ProjectionStatus::DELETING_INCL_EMITTED_EVENTS():
+                $this->delete(true);
+
+                return;
+            case ProjectionStatus::RESETTING():
+                $this->reset();
+                break;
+        }
```
Comment (sandrokeil) [inline src/Projection/PdoEventStoreReadModelProjection.php]: `default` case should be added, also if nothing is to do.

File: src/MySqlEventStore.php (hunk 1 above)
Comment (prolic) [inline src/MySqlEventStore.php]: Postgres has not UPDATE ... LIMIT. But we can remove it from MySqlEventStore for consistence reasons, anyway, there no possibility to update more than one row, as name is unique.

File: src/Projection/PdoEventStoreReadModelProjection.php (hunk 2 above)
Comment (prolic) [inline src/Projection/PdoEventStoreReadModelProjection.php]: done

File: src/MySqlEventStore.php (hunk 1 above)
Comment (sandrokeil) [inline src/MySqlEventStore.php]: Alright, let it as is. I was just wondering, because of my zero knowledge about postgres.

**Comment by codeliner** [comment]
that is really cool stuff. The event-store-ui will be a killer feature!

**Pull request merged**